File: src/patterns/creational/abstractfactory/cat.nim
type
    Cat* = ref object
        name: string
        specieName: string


proc newCat*(name: string, specieName: string): Cat =
    return Cat(
        name: name,
        specieName: specieName
    )

proc getName*(self: Cat): string =
    return self.name

proc setName*(self: Cat, name: string) =
    self.name = name

proc getSpecieName*(self: Cat): string =
    return self.specieName

proc setSpecieName*(self: Cat, specieName: string) =
    self.specieName = specieName

File: src/patterns/creational/abstractfactory/cat.go
package abstractfactory


type Cat struct {
	name string
	specieName string
}


func NewCat(name string, specieName string) *Cat {
	return &Cat{
		name: name,
		specieName: specieName,
	}
}

func (self *Cat) GetName() string {
	return self.name
}

func (self *Cat) SetName(name string) {
	self.name = name
}

func (self *Cat) GetSpecieName() string {
	return self.specieName
}

func (self *Cat) SetSpecieName(specieName string) {
	self.specieName = specieName
}